tests/test_kubernetes_tools.py

```diff
@@ -1681,9 +1719,27 @@ def test_get_pod_template_spec(
         mock_system_paasta_config.get_pod_defaults.return_value = dict(dns_policy="foo")
         mock_get_termination_grace_period.return_value = termination_grace_period
 
-        ret = self.deployment.get_pod_template_spec(
-            git_sha="aaaa123", system_paasta_config=mock_system_paasta_config
-        )
+        if autoscaling_metric_provider:
+            mock_config_dict = KubernetesDeploymentConfigDict(
+                min_instances=1,
+                max_instances=3,
+                autoscaling={"metrics_provider": autoscaling_metric_provider},
+                deploy_group="fake_group",
+            )
+            deployment = KubernetesDeploymentConfig(
+                service="kurupt",
+                instance="fm",
+                cluster="brentford",
+                config_dict=mock_config_dict,
+                branch_dict=None,
+            )
+            ret = deployment.get_pod_template_spec(
+                git_sha="aaaa123", system_paasta_config=mock_system_paasta_config
+            )
+        else:
+            ret = self.deployment.get_pod_template_spec(
```

Comment:
heh, took me a while to notice the difference between the two variable assignments here :p

Reply:
Made it more explicit as `autoscaled_deployment`